app/utils/Address.scala

```diff
@@ -0,0 +1,34 @@
+package utils
+
+import play.api.libs.json.{JsResult, JsString, JsValue, Reads, Writes}
+import repositories.PostgresProfile.api._
+
+case class Address(value: String) {
+  override def toString = value
+}
+
+object Address {
+  def apply(value: String) = new Address(value
+    .split(" - ")
+    .distinct
+    .map(_.trim)
+    .filterNot(_ == "FRANCE")
+    .filterNot(line => value.split(" - ").distinct.count(_.contains(line)) > 1)
```

Comment:
Peut-être sortir `value.split(" - ").distinct` dans une variable au départ, car on l'utilise plusieurs fois (une fois pour l'itération initiale puis une autre dans chaque itération — bon après tant qu'on n'a pas des adresses sur 100 lignes :) pas trop de souci de perf mais ça aide peut-être aussi la lisibilité)